examples/modules/fetch-newrelic-api-access-key/main.tf

```diff
@@ -0,0 +1,41 @@
+
+provider "graphql" {
+  url = var.fetch_access_keys_service.graphiql_url
+  headers = {
+    "Content-Type" = "application/json"
+    "API-Key" = var.fetch_access_keys_service.api_key != "" ? var.fetch_access_keys_service.api_key : var.create_access_keys_service.api_key
+  }
+}
+
+data "graphql_query" "basic_query" {
+  query_variables = {
+    "id"        = var.fetch_access_keys_service.key_id
+    "key_type"  = var.fetch_access_keys_service.key_type
+  }
+  query  = file("query.gql")
+  count = local.is_resource_created ? 0 : 1
+}
+
+resource "newrelic_api_access_key" "api_access_key" {
+  count  = var.create_access_keys_service.newrelic_account_id != "" ? 1 : 0
+  account_id  = var.create_access_keys_service.newrelic_account_id
+  key_type    = var.create_access_keys_service.key_type
+  ingest_type = var.create_access_keys_service.ingest_type
+  name        = "APM ${var.create_access_keys_service.key_type} License Key for ${var.create_access_keys_service.name}"
+  notes       = var.create_access_keys_service.notes
+}
+
+data "graphql_query" "query_with_id" {
+  query_variables = {
+    "id"        = newrelic_api_access_key.api_access_key[0].id
+    "key_type"  = var.create_access_keys_service.key_type
+  }
+  query = file("query.gql")
```

Comment:
@vinay-newrelic we seem to be running into a problem with using this module because of this reference of `query.gql`. I see the following error when I try using the sample configuration you've attached in the docs to use this module:

<img width="1392" alt="image" src="https://github.com/user-attachments/assets/cab50f19-038c-42f8-9475-77f2850f401c">


Not sure why this is a problem, it could be that modules don't support using files this way unless a path to the file is specified in the call of the module, maybe 🤔 

Anyway; to overcome this issue for now, I would suggest -
- placing the query in a multiline string (using the <<EOF heredoc syntax; a Google search can help explain how this can be used)
- put it inside `terraform.tfvars` as the value of the variable `query_string`
- declare a new variable `query_string` in variables.tf which can then be used as `var.query_string` instead of `file("query.gql")` everywhere else in the `main.tf` references.

Alternatively, you can also add `query_string` as a variable to `variables.tf`, use the multiline (heredoc) syntax to add the default value of the query to the variable (as the value of the argument default) and then use `var.query_string` everywhere `file("query.gql")` is currently being used.

Before you add the GraphQL query to the multiline string, please format it using any online GraphQL formatter and paste it back in the multiline query.